core/src/stark/air.rs

```diff
@@ -0,0 +1,198 @@
+    pub use crate::syscall::precompiles::weierstrass::WeierstrassDoubleAssignChip;
+    pub use crate::utils::ec::edwards::ed25519::Ed25519Parameters;
+    pub use crate::utils::ec::edwards::EdwardsCurve;
+    pub use crate::utils::ec::weierstrass::secp256k1::Secp256k1Parameters;
+    pub use crate::utils::ec::weierstrass::SWCurve;
+}
+
+/// An AIR for encoding RISC-V execution.
+///
+/// This enum contains all the different AIRs that are used in the Sp1 RISC-V IOP. Each variant is
+/// a different AIR that is used to encode a different part of the RISC-V execution, and the
+/// different AIR variants have a joint lookup argument.
+#[derive(MachineAir)]
+pub enum RiscvAir<F: PrimeField32> {
+    /// An AIR that containts a preprocessed program table and a lookup for the instructions.
+    Program(ProgramChip),
+    /// An AIR for the RISC-V CPU. Each row represents a cpu cycle.
+    Cpu(CpuChip),
+    /// An AIR for the RISC-V Add instruction.
+    Add(AddChip),
+    /// An AIR for the RISC-V Sub instruction.
+    Sub(SubChip),
+    /// An AIR for RISC-V Bitwise instructions.
+    Bitwise(BitwiseChip),
+    /// An AIR for RISC-V Mul instruction.
+    Mul(MulChip),
+    /// An AIR for RISC-V Div and Rem instructions.
+    DivRem(DivRemChip),
+    /// An AIR for RISC-V Lt instruction.
+    Lt(LtChip),
+    /// An AIR for RISC-V SLL instruction.
+    ShiftLeft(ShiftLeft),
+    /// An AIR for RISC-V SRL and SRA instruction.
+    ShiftRight(ShiftRightChip),
+    /// A lookup table for byte operations.
+    ByteLookup(ByteChip<F>),
+    /// An table for `less than` operation on field elements.
+    FieldLTU(FieldLTUChip),
+    /// A table for initializing the memory state.
+    MemoryInit(MemoryGlobalChip),
+    /// A table for finalizing the memory state.
+    MemoryFinal(MemoryGlobalChip),
+    /// A table for initializing the program memory.
+    ProgramMemory(MemoryGlobalChip),
+    /// A precompile for sha256 extend.
+    Sha256Extend(ShaExtendChip),
+    /// A precompile for sha256 compress.
+    Sha256Compress(ShaCompressChip),
+    /// A precompile for addition on the Elliptic curve ed25519.
+    Ed25519Add(EdAddAssignChip<EdwardsCurve<Ed25519Parameters>>),
+    /// A precompile for decompressing a point on the Edwards curve ed25519.
+    Ed25519Decompress(EdDecompressChip<Ed25519Parameters>),
+    /// A precompile for decompressing a point on the K256 curve.
+    K256Decompress(K256DecompressChip),
+    /// A precompile for addition on the Elliptic curve secp256k1.
+    Secp256k1Add(WeierstrassAddAssignChip<SWCurve<Secp256k1Parameters>>),
+    /// A precompile for doubling a point on the Elliptic curve secp256k1.
+    Secp256k1Double(WeierstrassDoubleAssignChip<SWCurve<Secp256k1Parameters>>),
+    /// A precompile for the Keccak permutation.
+    KeccakP(KeccakPermuteChip),
+    /// A precompile for the Blake3 compression function.
+    Blake3Compress(Blake3CompressInnerChip),
+}
+
+impl<F: PrimeField32> RiscvAir<F> {
```

Comment:
I feel like if you are going to go the `derive` route, then `included` should also be derived?